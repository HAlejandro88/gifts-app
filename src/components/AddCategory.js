import React,{ useState }  from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('')

    const handledinputChanged = (e) => {
        console.log(e, 'handledInputChange');
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(cats => [ inputValue, ...cats]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
        <p>{ inputValue }</p>
            <input type="text" value={ inputValue }  onChange={ handledinputChanged } />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


export default AddCategory
