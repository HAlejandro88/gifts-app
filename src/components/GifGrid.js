import React, { useState, useEffect } from 'react'

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&api_key=zWgr1JZwicNsrJtnNrmQlBorU4nTN2xZ';
        const request = await fetch(url);
        const { data } = await request.json();
        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
        setImages(gifs)
    }


    return (
        <div>
            <h3>{ category }</h3>
            <ol>
                {
                    images.map( ({ id, title }) => <li key={ id }>{ title }</li>)
                }
            </ol>
        </div>
    )
}
