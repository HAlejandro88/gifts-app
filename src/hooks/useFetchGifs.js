import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })


    useEffect(() => {
        getGifs(category)
              .then(imgs => {
                  setTimeout(() => {
                    setState({
                        data: imgs,
                        loading: false
                    })
                  }, 3000)
              })
    }, [category]) // evaluo unicamente cuando la categoria cambia


    return state
}