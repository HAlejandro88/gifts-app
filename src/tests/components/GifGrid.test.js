import React from 'react';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')


describe('Pruebas en el componente GigGrid', () => {

    const categoria = 'batman';
    test('comparar contra el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={ categoria } />)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost:3000/batman.jpg',
            title: 'Batman'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={ categoria } />)
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
        
    });
});