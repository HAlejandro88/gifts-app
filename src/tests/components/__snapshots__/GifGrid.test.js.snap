// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GigGrid comparar contra el snapshot 1`] = `
<Fragment>
  <h3>
    batman
  </h3>
  <p>
    Esta Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
